src/GL/rsp_gl.S

```diff
@@ -531,6 +548,8 @@ GLCmd_PreInitPipe:
     jal GL_UpdateTextureUpload
     nop
 
+rdpq_skiptexturing:
```

Comment:
(Continuing from above comment)
Instead, I would skip calling `GL_UpdateTextureUpload` depending on whether the texturing flag is set, like this:
```suggestion
    # * Skip if FLAG2_USE_RDPQ_TEXTURING is active

    andi t1, state_flags2, FLAG2_USE_RDPQ_TEXTURING
    beqz t1, GL_UpdateTextureUpload
    li ra, %lo(1f)
1:
```